extensions/amp-consent/0.1/test-e2e/test-amp-consent-server-side.js

```diff
@@ -0,0 +1,136 @@
+    // TODO (micajuineho): Add shadow-demo after #25985 is fixed and viewer-demo when...
+    environments: ['single'],
+  },
+  env => {
+    let controller;
+    let ui1;
+    let ui2;
+    let postPromptUi;
+    let tillResponded;
+    let accepted;
+    let autoReject;
+    let defaultBlock;
+    let notBlocked;
+    let twitter;
+    let requestBank;
+
+    beforeEach(() => {
+      controller = env.controller;
+      requestBank = env.requestBank;
+    });
+
+    async function findElements() {
+      ui1 = await controller.findElement('#ui1');
+      ui2 = await controller.findElement('#ui2');
+      postPromptUi = await controller.findElement('#postPromptUI');
+      tillResponded = await controller.findElement(
+        '[data-block-on-consent="_till_responded"]'
+      );
+      accepted = await controller.findElement(
+        '[data-block-on-consent="_till_accepted"]'
+      );
+      autoReject = await controller.findElement(
+        '[data-block-on-consent="_auto_reject"]'
+      );
+      defaultBlock = await controller.findElement(
+        '[data-block-on-consent="default"]'
+      );
+      notBlocked = await controller.findElement(
+        '[src="/examples/img/ima-poster.png"]'
+      );
+      twitter = await controller.findElement(
+        '[data-tweetid="885634330868850689"]'
+      );
+    }
+
+    async function verifyElementsBuilt(builtArray) {
+      const elements = [
+        tillResponded,
+        accepted,
+        autoReject,
+        defaultBlock,
+        notBlocked,
+        twitter,
+      ];
+
+      await expect(builtArray.length).to.equal(elements.length);
+
+      for (let i = 0; i < elements.length; i++) {
+        if (builtArray[i]) {
+          // Should be visible
+          await expect(
+            controller.getElementAttribute(elements[i], 'class')
+          ).to.not.match(/amp-notbuilt/);
+        } else {
+          // Should not be visible
+          await expect(
+            controller.getElementAttribute(elements[i], 'class')
+          ).to.match(/amp-notbuilt/);
+        }
+      }
+    }
+
+    async function verifyPromptsHidden(hiddenArray) {
+      const elements = [ui1, ui2, postPromptUi];
+
+      await expect(hiddenArray.length).to.equal(elements.length);
+
+      for (let i = 0; i < elements.length; i++) {
+        if (hiddenArray[i]) {
+          // Should be hidden
+          await expect(controller.getElementProperty(elements[i], 'hidden')).to
+            .be.true;
+        } else {
+          // Should not be hidden
+          await expect(controller.getElementProperty(elements[i], 'hidden')).to
+            .be.false;
+        }
+      }
+    }
+
+    it('should respect server side decision and persist it', async () => {
+      const currentUrl = await controller.getCurrentUrl();
+
+      // Block/unblock elements based off of 'reject' from response
+      await findElements();
+      await verifyElementsBuilt([true, false, true, false, true, false]);
```

Comment:
nit: instead of passing in an array, can we pass in an object that looks like
```
{
   'element': boolean
}
```
I feel that will make the test much more readable. Can the `element` can be the elementId which you can use `getElementById()` to get the element instance? Or you can host a map that maps the id to the element instance.